src/main/java/org/opensrp/service/ExportEventDataMapper.java

```diff
@@ -0,0 +1,192 @@
+import com.jayway.jsonpath.JsonPathException;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.opensrp.api.domain.Event;
+import org.opensrp.api.domain.Obs;
+import org.opensrp.domain.postgres.SettingsAndSettingsMetadataJoined;
+import org.opensrp.dto.ExportFlagProblemEventImageMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+
+import static org.opensrp.util.constants.EventDataExportConstants.OBS;
+import static org.opensrp.util.constants.EventDataExportConstants.FORM_SUBMISSION_FIELD;
+import static org.opensrp.util.constants.EventDataExportConstants.NOT_GOOD;
+import static org.opensrp.util.constants.EventDataExportConstants.MISUSE;
+import static org.opensrp.util.constants.EventDataExportConstants.SETTINGS_CONFIGURATION_EVENT_TYPE_TO_SETTINGS_IDENTIFIER;
+
+@Component
+public class ExportEventDataMapper {
+
+	@Autowired
+	private SettingService settingService;
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+	private static Logger logger = LoggerFactory.getLogger(ExportEventDataMapper.class.toString());
+
+	public List<Object> getExportEventDataAfterMapping(Object jsonObject, String eventType, boolean returnHeader,
+			boolean isSettingsExists) throws JsonProcessingException {
+		Map<String, String> columnNamesAndLabels = getColumnNamesAndLabelsByEventType(eventType);
+
+		String json = "";
+		if (jsonObject != null) {
+			json = objectMapper.writeValueAsString(jsonObject);
+		}
+
+		List<Object> headerData = new ArrayList<>();
+		List<Object> rowData = new ArrayList<>();
+		if (columnNamesAndLabels != null && columnNamesAndLabels.size() > 0 && returnHeader && isSettingsExists) {
+			for (Map.Entry<String, String> columnNameAndLabel : columnNamesAndLabels.entrySet()) {
+				headerData.add(columnNameAndLabel.getKey());
+			}
+			return headerData;
+		} else if (columnNamesAndLabels != null && !returnHeader && isSettingsExists) {
+			for (Map.Entry<String, String> columnNameAndLabel : columnNamesAndLabels.entrySet()) {
+				try {
+					Object fieldValue = JsonPath.read(json, columnNameAndLabel.getValue());
+					rowData.add(fieldValue);
+				}
+				catch (JsonPathException jsonPathException) {
+					logger.error("Key does not exists ", jsonPathException.getMessage());
+					rowData.add(null);
+				}
+			}
+			return rowData;
+		} else if (columnNamesAndLabels != null && !returnHeader && !isSettingsExists) {
+			Event event = objectMapper.readValue(json, Event.class);
+			for (Obs obs : event.getObs()) {
+				Object fieldValue = obs.getValues();
+				rowData.add(fieldValue);
+			}
+			return rowData;
+		} else { //for header without settings configurations
+			Event event = null;
+			if (!json.equals("")) {
+				event = objectMapper.readValue(json, Event.class);
+				for (Obs obs : event.getObs()) {
+					Object fieldValue = obs.getFormSubmissionField();
```

Comment:
you can also check `fieldCode` if `getFormSubmissionField` is null

Reply:
Done